src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder finalInfo = new StringBuilder(
+                String.format("Report for period %s - %s", dateFrom, dateTo))
+                .append(System.lineSeparator());
+
+        LocalDate from = LocalDate.parse(dateFrom, DATE_FORMAT);
+        LocalDate to = LocalDate.parse(dateTo, DATE_FORMAT);
+
+        for (int i = 0; i < names.length; i++) {
+            int salaryInfo = 0;
+            for (int j = 0; j < data.length; j++) {
+                if (data[j].contains(names[i])) {
+                    salaryInfo += parseLine(data[j], names[i], from, to);
+                }
+            }
+            finalInfo.append(names[i]).append(" - ").append(salaryInfo)
+                    .append(System.lineSeparator());
+        }
+
+        return finalInfo.toString().trim();
+    }
+
+    private int parseLine(String line, String name, LocalDate dateFrom, LocalDate dateTo) {
+        LocalDate date = LocalDate.parse(line.substring(0, 10), DATE_FORMAT);
```

Comment:
let's split the given string once and work with an array of parts, 
Now you are doing the same operation - searching for a particular index at least 6 times

also remember to use constants for indexes